x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilege.java

```diff
@@ -52,6 +52,19 @@
 import static org.elasticsearch.xpack.core.security.support.Automatons.patterns;
 import static org.elasticsearch.xpack.core.security.support.Automatons.unionAndMinimize;
 
+/**
+ * The name of an index related action always being with `indices:` followed by a sequence of slash-separated terms
+ * that generally describes the hierarchy (from broader to more specific) of the action. For example, the
+ * first level comprises `admin`, `monitor`, `data` which generally categorize an action into either an admin
+ * related function, or a monitoring related function or a user-data related function. Subsequent levels further
+ * narrow down the category until the meaning is specific enough.
+ *
+ * Note that these terms are meant to categorize what the action does, *not* how it should be invoked. This means
+ * whether an action is accessible via REST API should not contribute to its naming.
+ *
+ * Also note that the `internal:transport/proxy/` prefix is automatically added and stripped for actions that go
+ * through a CCR/CCS proxy. No action should be explicitly named like that.
+ */
```

Comment:
> However, I am asking you to add documentation to the code, e.g. in Constants.java you could add a code doc header explaining the different namespaces and purposes. Although this is test code, every new action has to be added here, so a developer would automatically stumble upon this documentation (or you find a better place and link it, whatever). Your explanations in the PR are great, but will be lost.

@hendrikmuhs I added this comment to `IndexPrivilege`. I hope this read right to you and potentially be useful in future. Thanks!